lib/gooddata/models/data_product.rb

```diff
@@ -0,0 +1,149 @@
+# encoding: UTF-8
+#
+# Copyright (c) 2010-2017 GoodData Corporation. All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+require_relative '../rest/resource'
+
+module GoodData
+  class DataProduct < Rest::Resource
+    include Mixin::UriGetter
+
+    attr_accessor :domain
+
+    ALL_DATA_PRODUCTS_PATH = '/gdc/domains/%{domain_name}/dataproducts'
+    ONE_DATA_PRODUCT_PATH = '/gdc/domains/%{domain_name}/dataproducts/%{id}'
+
+    class << self
+      def [](id, opts)
+        domain = opts[:domain]
+        fail ArgumentError, 'No :domain specified' if domain.nil?
```

Comment:
Refers to 'domain' more than self (maybe move it to another class?) (FeatureEnvy)